[Fix] Support HF models deployed with an OpenAI-compatible API.

heya5 opened the pull request:
## Motivation

I deploy a model with OpenAI-compatible server. But I get an error when running benchmarks in opencompass since it always load tokenizer with tiktoken.

## Modification

`hf_tokenizer` will be used when `tiktoken.encoding_for_model()` failed

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

tonysy commented:
Thanks for the contribution, would you like to update the PR to resolve the lint issues via pre-commit.?

heya5 commented:
> Thanks for the contribution, would you like to update the PR to resolve the lint issues via pre-commit.?

hi, I have already addressed the lint issues and updated the PR.

bittersweet1999 commented:
Thanks for your contribution!
